pkg/compact/downsample/streamed_block_writer.go

```diff
@@ -207,8 +209,14 @@ func (w *streamedBlockWriter) finalize() error {
 		return errors.Wrap(err, "sync blockDir")
 	}
 
+	if err := merr.Err(); err != nil {
```

Comment:
it's better to put it before [196](https://github.com/improbable-eng/thanos/pull/1070/files#diff-059ebb3a930d53936a164ef53a79d7f0R196) line to avoid wasting time/resources on writing index cache

Reply:
You are right. Do you have time for small PR on this? (: